### ISSUE TITLE
TypeError: (name || "").replace is not a function

### ISSUE THREAD

**Issue by dannyrb** [onury/docma#54]
![image](https://user-images.githubusercontent.com/5797588/48237781-4b233c00-e396-11e8-8041-df811c8343d6.png)

I'm able to generate documentation if I limit it to a small subset of my source. When I expand it to all files, I get this error. No new output is generated. I'm unable to identify where/what is throwing this error.

Any/all help is appreciated; even if it's just some advice on how to debug `docma` 👍

**Comment by onury** [comment]
my eyes! how can you even read that output :/ 

:) I suspect a non-string argument got passed to `cleanName()` function; which tries some replace operation on it. 

Can you replace `cleanName()` function  in `node_modules/jsdoc-x/src/lib/utils.js` with the code below to see the cause:
```js
function cleanName(name) {
    try {
        return (name || '').replace(/([^>]+>)?~?(.*)/, '$2')
            .replace(/^(module\.)?exports\./, '');
    } catch (err) {
        console.log('----> name:', name);
        console.log('----> type:', typeof name);
        throw err; // re-throw
    }
}
```

I think this can be fixed with:
```js
function cleanName(name) {
    return String(name || '').replace(/([^>]+>)?~?(.*)/, '$2')
        .replace(/^(module\.)?exports\./, '');
}
```

**Comment by dannyrb** [comment]
Thanks for the quick response, @onury!

New information logged:
```
----> name: 48
----> type: number
```

When using your suggested fix, I don't see error output in `PowerShell`, but when serving the output I get a blank screen and a console error:

![image](https://user-images.githubusercontent.com/5797588/48269473-a8a2a180-e405-11e8-8f56-c2c2c0d11246.png)

At some point, `helper.getSymbolInfo` is called with a `kind`, `scope`, and `asButton` of `undefined`. The `symbol` passed into `getSymbolData` looks like this:

```
$docmaLink: "?api#module:@public.waitForEnabledElementImageToLoad"
$kind: "constant"
$longname: "module:@public.waitForEnabledElementImageToLoad"
comment: "/**↵ * A promise that returns an EnabledElement who's image has loaded, or↵ * null if the provided element ceases being an enabledElement before an↵ * image has been loaded.↵ *↵ * @param {HTMLElement} element - An element that is an EnabledElement↵ * @returns {EnabledElement}↵ */"
description: "A promise that returns an EnabledElement who's image has loaded, or↵null if the provided element ceases being an enabledElement before an↵image has been loaded."
kind: "constant"
longname: "module:@public.waitForEnabledElementImageToLoad"
memberof: "module:@public"
meta:
code: {id: "astnode100069452", name: "exports.waitForEnabledElementImageToLoad", type: "VariableDeclaration"}
columnno: 0
filename: "wait.js"
lineno: 21
path: "C:\code\CornerstonePlayground\src\cornerstoneTools\src\util"
range: (2) [626, 1067]
__proto__: Object
name: "waitForEnabledElementImageToLoad"
params: Array(1)
0:
description: "An element that is an EnabledElement"
name: "element"
type:
names: ["HTMLElement"]
...
returns: Array(1)
0:
type:
names: ["EnabledElement"]
...
```

Because it has a `kind` of `constant`, it is not handled by any of the `if` statements in `getSymbolData` and `helper.getSymbolInfo` is called with 0 arguments passed. `helper.getSymbolInfo` looks like it will _always_ `throw` if `kind` is empty, so `helper.getSymbolInfo` should never be called without arguments?

If it helps, here is what the original code for that module looks like:

```js
import external from './../externalModules.js';

/**
 * Waits a set amount of time, then resolves. Can be chained off of to delay
 * next call in promise chain.
 *
 * @exports @public @function
 * @param {number} ms - number in ms to wait
 * @returns {Promise}
 */
export const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

/**
 * A promise that returns an EnabledElement who's image has loaded, or
 * null if the provided element ceases being an enabledElement before an
 * image has been loaded.
 *
 * @param {HTMLElement} element - An element that is an EnabledElement
 * @returns {EnabledElement}
 */
export const waitForEnabledElementImageToLoad = (element) => {
  try {
    const enabledElement = external.cornerstone.getEnabledElement(element);

    if (!enabledElement.image) {
      return wait(250).then(() => waitForEnabledElementImageToLoad(element));
    }

    // EnabledElement's image is loaded.
    return enabledElement;
  } catch (ex) {
    // Is no longer, or never was an enabled element, stop polling
    return null;
  }
};

export default wait;
```

**Comment by onury** [comment]
Thanks. I'm fixing this with some other revisions. I'll let you know.

**Comment by onury** [comment]
This is fixed now. 
I'm waiting for users on other issues before I release new version.

**Comment by onury** [comment]
Fixed with [Docma v3.0.0](https://onury.io/docma/changelog).

**Comment by warpdesign** [comment]
I am migrating an old template from v1 to v2 and stumbled upon this bug. Is there a workaround for this other than patching jsdoc-x?

Note: I am planning to move to the latest v3 version later but one step at a time :)
